update yml file

sin-sch opened the pull request:
Issue #14 and #15
Add Clippy and cargo fmt --check to yml file

Commit:
add clippy and cargo --fmt to yml file

Commit:
add websocket to clippy

Commit:
add cargofmt

FlorianUekermann commented on .github/workflows/rust.yml:
```diff
@@ -21,6 +21,9 @@ jobs:
       - run: cargo build --verbose --features ${{ matrix.crypto }},${{ matrix.ws }},${{ matrix.json }}
       - run: cargo test --verbose --features ${{ matrix.crypto }},${{ matrix.ws }},${{ matrix.json }}
-  
+      - run: cargo clippy --tests --no-default-features --features ${{ matrix.crypto }},${{ matrix.ws }}, ${{ matrix.json }}
```
remove extra space after comma

FlorianUekermann commented on .github/workflows/rust.yml:
```diff
@@ -21,6 +21,9 @@ jobs:
       - uses: actions/checkout@v3
       - run: rustup update ${{ matrix.toolchain }} && rustup default ${{ matrix.toolchain }}
+      - run: rustup component add clippy
+      - run: rustup component add cargofmt
```
I think it's `rustfmt`

FlorianUekermann commented on .github/workflows/rust.yml:
```diff
@@ -21,6 +21,9 @@ jobs:
+      - run: rustup component add clippy
+      - run: rustup component add cargofmt
+      - run: cargo fmt --check
       - run: cargo build --verbose --features ${{ matrix.crypto }},${{ matrix.ws }},${{ matrix.json }}
       - run: cargo test --verbose --features ${{ matrix.crypto }},${{ matrix.ws }},${{ matrix.json }}
```
I guess we should also add --no-default-features here.

sin-sch replied:
I think for compilation we need the `aws-lc-rs` feature.

FlorianUekermann reviewed (changes requested):
We need to actually fix the clippy warnings in this PR as well.

Commit:
update yml file

- delete extra space after comma
- change cargofmt to rustfmt
- add --no-default-features to add clippy, add rustfmt

Commit:
add --no-default-features

Commit:
remove --no-default-features

Commit:
Revert "remove --no-default-features"

This reverts commit 06a2c3df65a3a874a87eb44efe8dcbe70c609081.

Commit:
delete extra space

Commit:
fix clippy warnings

Commit:
fix clippy warnings in request_native

add Box<BodyEncodeState>

Commit:
edit matrix in .yml file
